client/blocks/reader-post-card/index.jsx

```diff
@@ -169,37 +83,24 @@ export default class RefreshPostCard extends React.Component {
 
 	render() {
 		const { post, site, feed, onCommentClick } = this.props;
-		const featuredImage = post.canonical_image;
 		const isPhotoOnly = post.display_type & DisplayTypes.PHOTO_ONLY;
 		const title = truncate( post.title, {
 			length: 140,
 			separator: /,? +/
 		} );
-
-		// only feature an embed if we know how to thumbnail & autoplay it
-		const featuredEmbed = head( filter( post.content_embeds, ( embed ) => embed.thumbnailUrl && embed.autoplayIframe ) );
-
-		// we only show a featured embed when all of these are true
-		//   - there is no featured image on the post that's big enough to pass as the canonical image
-		//   - there is an available embed
-		//
-		const useFeaturedEmbed = featuredEmbed &&
-			( ! featuredImage || ( featuredImage !== post.featured_image && featuredImage !== get( post, 'post_thumbnail.URL' ) ) );
-
-		const featuredAsset = useFeaturedEmbed
-			? <FeaturedVideo { ...featuredEmbed } videoEmbed={ featuredEmbed } />
-			: <FeaturedImage imageUri={ get( featuredImage, 'uri' ) } href={ post.URL } />;
+		const featuredAsset = ( <FeaturedAsset post={ post } /> );
+		const hasFeaturedAsset = !! featuredAsset;
 
 		const classes = classnames( 'reader-post-card', {
-			'has-thumbnail': !! featuredAsset,
+			'has-thumbnail': hasFeaturedAsset,
 			'is-photo': isPhotoOnly
 		} );
 
 		return (
 			<Card className={ classes } onClick={ this.handleCardClick }>
 				<PostByline post={ post } site={ site } feed={ feed } />
 				<div className="reader-post-card__post">
-					{ featuredAsset }
+					{ hasFeaturedAsset && featuredAsset }
```

Comment:
Correct! I was trying to chase down a problem where `FeaturedAsset` was handing back `undefined` sometimes. Will be removing `hasFeaturedAsset` before review :)